### PR TITLE
The route requires a stock location to be specified

### PR DESCRIPTION
**Pull request by richardnuno**

### CODE REVIEW

**Comment by athal7** [comment]
:+1:

**Comment by jordan-brough** [comment]
I'm totally confused about how the `payments_api` one manages to work since it requires a param also.  `order_id` seems to get set as a default url option, though I'm not sure how that happens...?

Both of the argument-required routes seem really indirect and hard to follow though.  Would it be a big hassle to maybe move this route to [here](https://github.com/bonobos/spree/blob/becc0a106279fb3e3431a019a5282ffc75924052/core/app/views/spree/shared/_routes.html.erb) and make it like the `apply_coupon_code` one maybe?  I don't know much about how these are all used but that seems slightly better.

**Comment by gmacdougall** [comment]
This change looks fine, but it's difficult for me to know if it does what you're trying to do without knowing what you're trying to do. Could you please provide a description in the PR?

Thanks!

**Comment by richardnuno** [comment]
@gmacdougall good point. I came across this issue when trying to make some enhancements to the stock management ui. I was adding the ability to filter the displayed stock items by a selected stock_location_id. Since the stock_location_id was now part of the `params` it broke the assignment. Closing this PR for now to look into @jordan-brough's suggested approach.
